src/coreclr/jit/morphblock.cpp

```diff
@@ -1395,7 +1415,7 @@ GenTree* MorphCopyBlockHelper::CopyFieldByField()
                     }
                     else
                     {
-                        if (i == (fieldCnt - 1))
+                        if (numCreated == fieldCnt - dyingFieldCnt)
```

Comment:
It looks like `numCreated` is only used here for the "last field" check.

This check can now actually be simplified to (once again) use the first field instead (`result == nullptr`), it was originally altered to protect against bad zero-offset field sequences.

Reply:
Changed this and the other check to `result == nullptr`